data/data.go

```diff
@@ -34,15 +35,38 @@ func NewSelectorOrigin() *SelectorOrigin {
 
 func (s *SelectorOrigin) Parse(parser structure.ObjectParser) {
 	s.ID = parser.String("id")
+	s.Time = parser.String("time")
 }
 
 func (s *SelectorOrigin) Validate(validator structure.Validator) {
 	validator.String("id", s.ID).Exists().NotEmpty().LengthLessThanOrEqualTo(SelectorOriginIDLengthMaximum)
+	validator.String("time", s.Time).AsTime(time.RFC3339Nano).NotZero()
+}
+
+func (s *SelectorOrigin) Includes(other *SelectorOrigin) bool {
```

Comment:
Feels worth a test with so many decisions in this? Or is it somehow covered elsewhere

Reply:
Added tests.